src/ratbag_emu/protocol/base.py

```diff
@@ -127,6 +126,7 @@ def simulate_action(self, actions):
 
             # XY movement
             if action['action']['type'] == 'xy':
+                assert self.hw_profile.get_dpi_value(), 'The device needs a DPI value to emulate mouse movement'
```

Comment:
Having an `assert` in a normal code can be rather dangerous. You should probably rely on an earlier catch of the error

Reply:
We don't know if the client will want to do mouse movement before. For ex. if the client is emulating a mouse everything will be fine until he tries to do mouse movement. The code that calls `simulate_action` has a try block to deal with the exception.

Maybe I can just skip that action instead of having an assert, but I think this is more verbose and prevents errors in the client.

Reply:
Other solution is to consider defaults for each value. So if I ask a mouse with:
```json
json = '''
    {
      "profiles": [
        { "is_active": true }
      ]
    }'''
```

Why would you prevent me to compute the dpi from evdev?

Reply:
We might want to describe devices that do not have those properties. For example, I might want to emulate a keyboard.

Reply:
it really doesn't matter.
The `generic` device can be a set of a mouse and a keyboard (so you can send macros too). So you don't need to worry about keyboards  IMO.

Also, note that I am not asking you to be more restrictive, but less strict. If you have a default device with random default values, you are not preventing a user to input random commands. It's then the user responsibility to have something sane.